src/tests/end2end/BatchNormTests.cpp

```diff
@@ -20,146 +20,140 @@ class BatchNormTests : public WebnnTest {
     }
 
   protected:
-    void CheckBatchNorm(const std::vector<int32_t>& inputShape,
-                        const std::vector<float>& inputData,
-                        const std::vector<float>& meanData,
-                        const std::vector<int32_t>& meanShape,
-                        const std::vector<float>& varianceData,
-                        const std::vector<int32_t>& varianceShape,
+    struct Tensor {
+        std::vector<int32_t> shape;
+        std::vector<float> value;
+    };
+    std::vector<SHARED_DATA_TYPE> mConstants;
+    void CheckBatchNorm(const Tensor& input,
+                        const Tensor& mean,
+                        const Tensor& variance,
                         const std::vector<float>& expectedValue,
-                        const ml::BatchNormOptions* options = nullptr) {
-        const ml::Operand a = utils::BuildInput(builder, "a", inputShape);
-        const ml::Operand mean = utils::BuildConstant(builder, meanShape, meanData.data(),
-                                                      meanData.size() * sizeof(float));
-        const ml::Operand variance = utils::BuildConstant(
-            builder, varianceShape, varianceData.data(), varianceData.size() * sizeof(float));
-        const ml::Operand b = builder.BatchNorm(a, mean, variance, options);
-        const ml::Graph graph = utils::Build(builder, {{"b", b}});
+                        const Tensor& scale = {},
+                        const Tensor& bias = {},
+                        ml::BatchNormOptions options = {},
+                        utils::FusedActivation activation = utils::FusedActivation::NONE) {
+        const ml::Operand x = utils::BuildInput(builder, "input", input.shape);
+        const ml::Operand m = utils::BuildConstant(builder, mean.shape, mean.value.data(),
+                                                   mean.value.size() * sizeof(float));
+        const ml::Operand v = utils::BuildConstant(builder, variance.shape, variance.value.data(),
+                                                   variance.value.size() * sizeof(float));
+        if (!scale.value.empty()) {
+            options.scale = utils::BuildConstant(builder, scale.shape, scale.value.data(),
+                                                 scale.value.size() * sizeof(float));
+        }
+        if (!bias.value.empty()) {
+            options.bias = utils::BuildConstant(builder, bias.shape, bias.value.data(),
+                                                bias.value.size() * sizeof(float));
+        }
+        if (activation != utils::FusedActivation::NONE) {
+            options.activation = utils::createActivationOperator(mConstants, builder, activation);
```

Comment:
Would you like to test `createActivationOperand` as well?

Reply:
Any responses?

Reply:
I just align with [polyfill batchNorm tests](https://github.com/webmachinelearning/webnn-polyfill/pull/98/files?file-filters%5B%5D=.js#diff-32905e9e5b2ba197c3a21a7c6926bce98089d9e64823ffe676b0eba231496cd0) here. Do you mean I need to test createActivationOperand for unfused BatchNorm ?

Reply:
The alignment is fine. We can add it later.

Reply:
Thanks, I think so. And should we add some tests for fused BatchNorm with sigmoid/relu6?

Reply:
It should, however as they are not common usages, could you or @BruceDai file an issue to track the missing of test cases?